src/main/java/reposense/commits/CommitInfoAnalyzer.java

```diff
@@ -93,9 +105,73 @@ public static CommitResult analyzeCommit(CommitInfo commitInfo, RepoConfiguratio
             extractTagNames(tags);
         }
 
-        int insertion = getInsertion(statLine);
-        int deletion = getDeletion(statLine);
-        return new CommitResult(author, hash, date, messageTitle, messageBody, tags, insertion, deletion);
+        if (statLine.isEmpty()) { // empty commit, no files changed
+            return new CommitResult(author, hash, date, messageTitle, messageBody, tags, 0, 0, new HashMap<>());
```

Comment:
consider adding a new constructor that only takes in `CommitResult(author, hash, date, messageTitle, messageBody, tags)` to handle the zero/default values.

Reply:
Noted.